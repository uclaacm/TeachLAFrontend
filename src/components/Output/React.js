import { getJsSrcDocLoggingScript } from "./constants";

const getReactSrcDocHead = () => `
    <head>
        <style>html,body: {margin:0, width:100%}</style>
        <script src="https://unpkg.com/@babel/standalone/babel.min.js"></script>
        <script src="https://unpkg.com/react@17/umd/react.development.js" crossorigin></script>
        <script src="https://unpkg.com/react-dom@17/umd/react-dom.development.js" crossorigin></script>
        <script src="https://unpkg.com/@material-ui/core@latest/umd/material-ui.development.js" crossorigin="anonymous"></script>
        <style>
          #inner {
            height:100px;
            background-color:#222;
            color: #DDD;
            font-family: monospace;
            word-wrap:break-word;
            overflow:auto;
            margin: 10px auto;
            position:relative;
            padding: 10px 35px 10px 10px;
            width: 100%;
            box-sizing: border-box;         /* For IE and modern versions of Chrome */
            -moz-box-sizing: border-box;    /* For Firefox                          */
            -webkit-box-sizing: border-box; /* For Safari                           */
          }
          #output {
            display: block;
            position: relative;
            background: white;
            min-height: 70vh;
            padding: 10px 35px 10px 10px;
          }
          #closeConsoleButton { position: fixed; top: 20px; right: 30px; color: #ddd;}
        </style>
    </head>
  `;

const getUserScript = (code) => `
  <script type="text/babel">
    ${code}
    ReactDOM.render(
      <App />,
      document.getElementById('output'),
    );
  </script>
`;

const getReactSrcDocBody = (code, showConsole) => `
    <body>
      ${
        showConsole
          ? `<div id="outer"><textarea id="inner"></textarea></div>`
          : `<div id="outer" style="display:none;"><textarea id="inner"></textarea></div>`
      }
      ${getJsSrcDocLoggingScript()}
      ${getUserScript(code)}
      <div id="output"></div>
    </body>
  `;

export default function (code, showConsole) {
  return `<html> ${getReactSrcDocHead()} ${getReactSrcDocBody(code, showConsole)}</html>`;
}
