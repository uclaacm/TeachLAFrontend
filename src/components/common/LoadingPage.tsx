import { useState, useEffect } from 'react';
import { RingLoader } from 'react-spinners';
import '../../styles/Loading.scss';
import { GH_REPO_NAME } from '../../constants';

interface LoadingProps {
  showHelpText: boolean;
}

const Loading = function ({ showHelpText = false } : LoadingProps) {
  const [hasHelpText, setHasHelpText] = useState(showHelpText);

  const timer = setTimeout(
    () => {
      setHasHelpText(true);
    },
    2000,
  );

  useEffect(() => () => {
    clearTimeout(timer);
  }, []);

  return (
    <div className="Loading">
      <div className="Loading-title">Loading</div>
      <RingLoader color="#171124" size={250} loading />
      { hasHelpText && (
        <p className="Loading-page-text" style={{ color: 'white' }}>
          Looks like loading is taking a bit long! If it takes too long, submit an issue on
          {' '}
          <span> </span>
          <a href={`${GH_REPO_NAME}/issues`} className="Loading-link-text">
            Github
          </a>
          .
        </p>
      )}
    </div>
  );
};

export default Loading;
