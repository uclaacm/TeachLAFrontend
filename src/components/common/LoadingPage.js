import React from "react";
import { RingLoader } from "react-spinners";
import "styles/Loading.scss";
import { GH_REPO_NAME } from "../constants";

/*
	Props:
		bgColor: string representing the color of the background of the img (can be hex color, rgb(r, g, b, a), or color name)
		textColor: string representing the color of the text in the button (can be hex color, rgb(r, g, b, a), or color name)
		imgSrc: string representing the location of the img used as the icon (can be in the form of URL, path location, or data representing image)
		textPadding: string representing padding to the left of the text, i.e. distance from the img (give px units)
*/

class Loading extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      showText: false,
      timer: 0,
    };
  }
  componentDidMount = () => {
    this.setState.timer = setTimeout(() => {
      this.setState({ showText: true });
    }, 2000);
  };

  componentWillUnmount = () => {
    clearTimeout(this.setState.timer);
  };

  render = () => {
    return (
      <div className="Loading">
        <div className="Loading-title">Loading</div>
        <RingLoader color={"#171124"} size={250} loading={true} />
        {this.state.showText && (
          <p className="Loading-page-text" style={{ color: "white" }}>
            Looks like loading is taking a bit long! If it takes too long, submit an issue on{" "}
            <span> </span>
            <a href={`${GH_REPO_NAME}/issues`} className="Loading-link-text">
              Github
            </a>
            .
          </p>
        )}
      </div>
    );
  };
}

export default Loading;
