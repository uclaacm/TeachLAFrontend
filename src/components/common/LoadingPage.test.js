import { shallow, mount } from 'enzyme';
import React from 'react';
import LoadingPage from './LoadingPage';

describe('LoadingPage', () => {
  it('smoke test', () => {
    const component = shallow(<LoadingPage />);
    expect(component.exists()).toBe(true);
  });

  it('should render correctly', () => {
    const component = shallow(<LoadingPage />);

    expect(component).toMatchSnapshot();
  });

  it('should handle showHelpText prop properly', () => {
    const component = shallow(<LoadingPage />);
    expect(component).toMatchSnapshot();

    const component_sht_false = shallow(<LoadingPage showHelpText={false} />);
    expect(component_sht_false.find('.Loading-page-text')).toHaveLength(0);

    const component_sht_true = shallow(<LoadingPage showHelpText />);
    expect(component_sht_true.find('.Loading-page-text')).toHaveLength(1);
  });

  it('shows the help text after 2 seconds', async () => {
    // TODO - use mock timers instead? didn't work the first time we tried it :(
    const component = shallow(<LoadingPage />);
    expect(component.find('.Loading-page-text')).toHaveLength(0);
    await new Promise((r) => setTimeout(r, 2000));
    expect(component.find('.Loading-page-text')).toHaveLength(1);
  });
});
