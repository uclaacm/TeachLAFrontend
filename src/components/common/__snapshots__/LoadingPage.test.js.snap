// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingPage should handle showHelpText prop properly 1`] = `
<div
  className="Loading"
>
  <div
    className="Loading-title"
  >
    Loading
  </div>
  <Loader
    color="#171124"
    css=""
    loading={true}
    size={250}
  />
</div>
`;

exports[`LoadingPage should render correctly 1`] = `
<div
  className="Loading"
>
  <div
    className="Loading-title"
  >
    Loading
  </div>
  <Loader
    color="#171124"
    css=""
    loading={true}
    size={250}
  />
</div>
`;
